prime-dotnet-webapi/Controllers/ProvisionerAccessController.cs

```diff
@@ -179,14 +213,20 @@ public async Task<ActionResult> GetGpidDetail()
         [ProducesResponseType(typeof(ApiResultResponse<IEnumerable<HpdidLookup>>), StatusCodes.Status200OK)]
         public async Task<ActionResult> HpdidLookup([FromQuery] string[] hpdids)
         {
+            var hpdidsJson = JsonConvert.SerializeObject(hpdids);
+            var logId = await _vendorAPILogService.CreateLogAsync(User.GetPrimeUsername(), "api/provisioner-access/gpids", hpdidsJson);
             if (hpdids != null && hpdids.Length > _hpdidLimit_HpdidLookup)
             {
-                return BadRequest($"number of {nameof(hpdids)} should not exceed {_hpdidLimit_HpdidLookup}");
+                var errorMessage = $"number of {nameof(hpdids)} should not exceed {_hpdidLimit_HpdidLookup}";
+                await _vendorAPILogService.UpdateLogAsync(logId, null, errorMessage);
+                return BadRequest(errorMessage);
             }
             else
             {
                 var result = await _enrolleeService.HpdidLookupAsync(hpdids);
+                var resultJson = JsonConvert.SerializeObject(result);
```

Comment:
Can what is stored in database have keys in same case as actual output?  e.g. 
```
[
        {
            "hpdid": "46fjuzcic6vdp4dpmacvwrctn3kzrmne",
            "gpid": "K?@?74HN$FA33YGN7*6E",
            "status": "Indefinite absence",
            "accessType": null,
            "licences": null
        }
]
```
rather than 
```
[
	{
		"Hpdid": "46fjuzcic6vdp4dpmacvwrctn3kzrmne",
		"Gpid": "K?@?74HN$FA33YGN7*6E",
		"Status": "Indefinite absence",
		"AccessType": null,
		"Licences": null
	}
]
```

Reply:
fixed